### PR TITLE
Improve fake executor to handle coin::transfer and aptos_account::transfer as well

### PR DESCRIPTION
**Pull request by igor-aptos**

### CODE REVIEW

File: execution/executor-benchmark/src/fake_executor.rs [hunk 1]
```diff
@@ -46,26 +52,13 @@ impl FakeExecutor {
...
-            vec![],
-            /*gas_used=*/ 1,
-            TransactionStatus::Keep(ExecutionStatus::Success),
-        ))
-    }
-
-    fn handle_account_creation(
+    fn handle_account_creation_and_transfer(
         sender_address: AccountAddress,
         new_account_address: AccountAddress,
```
Comment (msmouse) [inline execution/executor-benchmark/src/fake_executor.rs:57]: let's call it `receiver_address`? We don't require or assume it to be "new".

File: execution/executor-benchmark/src/fake_executor.rs [hunk 2]
```diff
@@ -142,20 +106,88 @@ impl FakeExecutor {
...
+
+            let mut new_coin_store = {
+                let _timer = TIMER
+                    .with_label_values(&["read_new_coin_store"])
+                    .start_timer();
+                DbAccessUtil::get_coin_store(&new_coin_store_key, state_view)?.unwrap()
+            };
+
+            if initial_balance != 0 {
+                new_coin_store.coin += initial_balance;
```
Comment (msmouse) [inline execution/executor-benchmark/src/fake_executor.rs:140]: nit "=" instead of "+=" is cleaner, unless it's deliberate, where a comment is desired.

**Review by msmouse — APPROVED** [review]

**Comment by igor-aptos** [comment]
@msmouse addressed your comments, with doing a bit more of a cleanup, and added support for batch_transfer

File: execution/executor-benchmark/src/fake_executor.rs [hunk 3]
```diff
@@ -106,110 +122,207 @@ impl FakeExecutor {
...
+        state_view: &CachedStateView,
+        fail_on_existing: bool,
+        fail_on_missing: bool,
+    ) -> Result<TransactionOutput> {
+        let _timer = TIMER.with_label_values(&["account_creation"]).start_timer();
+
+        let mut output = {
+            let output = Self::withdraw_from_signer(sender_address, transfer_amount, state_view)?;
+            match output {
+                Ok(output) => output,
```
Comment (msmouse) [inline execution/executor-benchmark/src/fake_executor.rs:250]: use `Option::or_else()`?

**Review by zekun000 — APPROVED** [review]

File: execution/executor-benchmark/src/fake_executor.rs (hunk 3 above)
Comment (igor-aptos) [reply execution/executor-benchmark/src/fake_executor.rs:250]: unfortunately, I return from the top level function in the "else", so I cannot do that easily

a little bit too nested with Result<Result<IncrementalOutput, TransactionStatus>>, so having two different Err types basically

**Pull request merged**